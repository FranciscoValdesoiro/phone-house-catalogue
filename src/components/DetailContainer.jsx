import React from 'react';
import DetailComponent from './DetailComponent';
import { setSelectedPhoneAction } from '../redux/ducks/phones';
import {useDispatch, useSelector} from 'react-redux'
import {useHistory} from 'react-router-dom'

const DetailContainer = () => {

    const dispatch = useDispatch()
    const setPhone = (phone) => dispatch(setSelectedPhoneAction(phone))
    const history = useHistory()

    const phone = useSelector(state => state.phones.PhoneSelected)

    return (
        <DetailComponent phone={phone} setPhone={setPhone} history={history}/>
    )           
}

export default DetailContainer;
