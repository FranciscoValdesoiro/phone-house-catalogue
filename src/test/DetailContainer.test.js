import React from 'react';
import { mount } from "enzyme";
import DetailContainer from "../components/DetailContainer";
import '@testing-library/jest-dom'
import {useSelector} from 'react-redux'
import mockPhones from '../__mocks__/phones'

jest.mock('react-redux', () => ({
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
  }));

describe('DetailContainer', () => {
  
    const [phone] = mockPhones;
    
    useSelector.mockImplementation((selector) => selector({
      phones: {
          PhoneList: [],
          PhoneSelected: phone
      }
    }));

    const setPhone = jest.fn()
    const history = { goBack: jest.fn()}

    const wrapper = mount(<DetailContainer phone={phone} setPhone={setPhone} history={history} />)

    it('Should render the DetailContainer component', () => {
      expect( wrapper ).toMatchSnapshot()
    })

  });