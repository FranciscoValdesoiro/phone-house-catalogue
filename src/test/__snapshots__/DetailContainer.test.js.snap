// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailContainer Should render the DetailContainer component 1`] = `
<div
  className="card-container-detail"
  data-cy="card-details"
>
  <img
    alt="iPhone 7phone image"
    className="phone-image"
    src="/images/IPhone_7.png"
  />
  <div
    className="details-container"
  >
    <h2>
      iPhone 7
    </h2>
    <p>
      iPhone 7 dramatically improves the most important aspects of the iPhone experience. It introduces advanced new camera systems. The best performance and battery life ever in an iPhone. Immersive stereo speakers. The brightest, most colorful iPhone display. Splash and water resistance*. And it looks every bit as powerful as it is. This is iPhone 7.
    </p>
    <p>
      Color: 
      black
    </p>
    <p>
      Screen: 
      4,7 inch IPS
    </p>
    <p>
      Manufacturer: 
      Apple
    </p>
    <p>
      Processor: 
      A10 Fusion
    </p>
    <p>
      Ram: 
      2
    </p>
    <p>
      Price: 
      769
      $
    </p>
    <button
      data-cy="btn-go-back"
      onClick={[Function]}
    >
       Go back
    </button>
  </div>
</div>
`;
